app/src/main/java/meme5/c4q/nyc/meme_project/MainActivity.java

```diff
@@ -1,15 +1,104 @@
 package meme5.c4q.nyc.meme_project;
 
 import android.app.Activity;
-import android.os.Bundle;
 
+import android.content.Context;
+import android.content.Intent;
+import android.database.Cursor;
+import android.graphics.Bitmap;
+import android.net.Uri;
+import android.provider.MediaStore;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Toast;
+import java.io.File;
+import java.io.FileOutputStream;
 
 public class MainActivity extends Activity {
 
+    private static final int REQUEST_IMAGE_CAPTURE = 1;
+    private Bitmap thumbnail;
+    private String imgFilePath;
+    private String thumbFileName;
+    private boolean imageSelected;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        imageSelected = false;
+        Log.d("onCreate Launched", "onCreate");
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        Log.d("onResume Launched", "onResume");
+        if(imageSelected){
+            launchChooseMeme();
+        }
+    }
+
+    private void launchChooseMeme(){
+        Intent chooseMeme = new Intent(this,ChooseMemeStyle.class);
+        chooseMeme.putExtra("thumbnailFileName", thumbFileName);
```

Comment:
Tasha wanted a thumbnail for her chooseMemeStyle activity so I added it there. You can also take the full resolution image and shrink it so it's not slowing down your activity but that's extra unnecessary code cause the camera intent already provides the low res version